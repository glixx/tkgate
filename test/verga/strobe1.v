module top;
  integer i,a;

  initial
    begin
	$monitor("%d: $monitor a=%d",$time,a);
	for (i = 0;i < 20;i = i + 1)
	  begin
	    if ( (i & 1) == 0)
	      a <= i;

	    if (i == 8) $monitoroff;
	    if (i == 14) $monitoron;
	    if (i == 16) $monitor("%d: $monitor(2) 3*a=%d",$time,3*a);

	    $display("%d: $display 2*a=%d",$time,2*a);
	    $strobe("%d: $strobe 2*a=%d",$time,2*a);
	    #1;
	  end
    end
endmodule
