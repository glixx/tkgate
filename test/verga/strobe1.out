0: $display 2*a=x
0: $strobe 2*a=0
0: $monitor a=0
1: $display 2*a=0
1: $strobe 2*a=0
2: $display 2*a=0
2: $strobe 2*a=4
2: $monitor a=2
3: $display 2*a=4
3: $strobe 2*a=4
4: $display 2*a=4
4: $strobe 2*a=8
4: $monitor a=4
5: $display 2*a=8
5: $strobe 2*a=8
6: $display 2*a=8
6: $strobe 2*a=12
6: $monitor a=6
7: $display 2*a=12
7: $strobe 2*a=12
8: $display 2*a=12
8: $strobe 2*a=16
9: $display 2*a=16
9: $strobe 2*a=16
10: $display 2*a=16
10: $strobe 2*a=20
11: $display 2*a=20
11: $strobe 2*a=20
12: $display 2*a=20
12: $strobe 2*a=24
13: $display 2*a=24
13: $strobe 2*a=24
14: $display 2*a=24
14: $strobe 2*a=28
14: $monitor a=14
15: $display 2*a=28
15: $strobe 2*a=28
16: $display 2*a=28
16: $strobe 2*a=32
16: $monitor(2) 3*a=48
17: $display 2*a=32
17: $strobe 2*a=32
18: $display 2*a=32
18: $strobe 2*a=36
18: $monitor(2) 3*a=54
19: $display 2*a=36
19: $strobe 2*a=36
